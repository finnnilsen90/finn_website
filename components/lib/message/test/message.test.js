const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Message = require('../message.js');

let newMessage = new Message();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Message).toBeDefined();
    })

})  
    
describe('<Message />', () => {

    const wrapper = render(<Message />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });


})

describe('message state', () => {
    let msg = 'This is an error message.'
    const wrapper = mount(<Message state='error' class='test_class' message={msg}/>, { attachTo: document.body });

    
    it('mount components', () => {
        expect(wrapper).toMatchSnapshot();
    });


    it('props should make state error', () => {
        expect(wrapper.props().state).toEqual('error');
    })

      it('props should output error message', () => {
        expect(wrapper.props().message).toEqual(msg);
    })

    it('state should make error class', () => {
        expect(wrapper.find('.error')).toHaveLength(1)
    })

    it('should find class', () => {
        expect(wrapper.find('.test_class')).toHaveLength(1)
    })
})