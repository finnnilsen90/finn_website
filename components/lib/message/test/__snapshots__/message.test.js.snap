// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders components 1`] = `
<div
  class="message_Container undefined"
>
  <div
    class="msg_box undefined"
  >
    <p />
  </div>
</div>
`;

exports[`message state mount components 1`] = `
<Message
  class="test_class"
  message="This is an error message."
  state="error"
>
  <div
    className="message_Container test_class"
  >
    <div
      className="msg_box error"
    >
      <p>
        This is an error message.
      </p>
    </div>
  </div>
</Message>
`;
